client/shared/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -289,16 +291,64 @@ object Client {
       go(source).stream
     }
 
-    def runOrDispose(req: Request[F]): F[Resource[F, Response[F]]] =
-      Ref[F].of(false).map { disposed =>
-        val req0 = addHostHeaderIfUriIsAbsolute(req.pipeBodyThrough(until(disposed)))
-        Resource
-          .eval(app(req0))
-          .onFinalize(disposed.set(true))
-          .map(_.pipeBodyThrough(until(disposed)))
-      }
+    def run(
+        message: Message[F],
+        refOp: Option[Ref[F, Boolean]] = None,
+    ): Resource[F, Response[F]] = message match {
+      case req: Request[F] =>
+        Resource.eval(Ref[F].of(false)).flatMap { disposed =>
+          val reqAugmented =
+            addHostHeaderIfUriIsAbsolute(req.pipeBodyThrough(until(disposed)))
+          Resource
+            .eval(app(reqAugmented))
+            .onFinalize(disposed.set(true))
+            .flatMap(run(_, Option(disposed)))
+        }
+      case resp: Response[F] =>
+        for {
+          disposed <- Resource.eval(refOp.fold(Ref[F].of(false))(F.pure))
+          channel <- Resource.eval(Channel.synchronous[F, Byte])
+
+          _ <- resp.body
+            .evalMap(channel.send)
+            .takeWhile(_.isRight)
+            .onFinalize(channel.close.void)
```

Comment:
Isn't this the definition of `channel.sendAll` ? Why copy it here?
https://github.com/typelevel/fs2/blob/509d74fdcc5bf0735cabcf829d0a5ba21a949ed9/core/shared/src/main/scala/fs2/concurrent/Channel.scala#L147-L151

Reply:
Actually, I'm afraid this is sending byte-by-byte instead of chunk-by-chunk. We should probably change this to work in terms of chunks.

Reply:
> Isn't this the definition of channel.sendAll
I changed the implementation so often that I did not notice that I repeated the same thing. I've replaced it with sendAll.

> sending byte-by-byte instead of chunk-by-chunk
done